File: lib/core/src/models/speaker.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'speaker.freezed.dart';
part 'speaker.g.dart';

@freezed
class Speaker with _$Speaker {
  @JsonSerializable(explicitToJson: true)
  const factory Speaker({
    required String id,
    required String firstName,
    required String lastName,
    required String fullName,
    required String bio,
    required String tagLine,
    required String profilePicture,
    required bool isTopSpeaker,
    required List<dynamic> links,
    required List<int> sessions,
  }) = _Speaker;

  factory Speaker.fromJson(Map<String, Object?> json) =>
      _$SpeakerFromJson(json);
}

File: conference_app_toolkit/__brick__/firebase/functions/src/interfaces/speaker.ts
export interface Speaker {
  id: string;
  firstName: string;
  lastName: string;
  fullName: string;
  bio: string;
  tagLine: string;
  profilePicture: string;
  isTopSpeaker: boolean;
  links: Array<Record<string, any>>;
  sessions: Array<number>;
  questionAnswers?: Array<Record<string, any>>;
}

